sdk/eventhub/Azure.Messaging.EventHubs.Processor/src/EventProcessorClient.cs

```diff
@@ -712,57 +712,55 @@ protected EventProcessorClient() : base()
-        /// <param name="context">The context of the partition the checkpoint is associated with.</param>
+        /// <param name="partitionId">The identifier of the partition to associate with the checkpoint.</param>
+        /// <param name="eventData">The event on which to base the checkpoint.  When using the checkpoint, a processor will start reading from the next available event in the stream.</param>
         /// <param name="cancellationToken">A <see cref="CancellationToken" /> instance to signal the request to cancel the operation.</param>
         ///
-        internal Task UpdateCheckpointAsync(EventData eventData,
-                                            PartitionContext context,
-                                            CancellationToken cancellationToken)
+        protected internal virtual async Task UpdateCheckpointAsync(string partitionId,
+                                                                    EventData eventData,
+                                                                    CancellationToken cancellationToken)
         {
             cancellationToken.ThrowIfCancellationRequested<TaskCanceledException>();
 
+            Argument.AssertNotNullOrEmpty(partitionId, nameof(partitionId));
             Argument.AssertNotNull(eventData, nameof(eventData));
             Argument.AssertInRange(eventData.Offset, long.MinValue + 1, long.MaxValue, nameof(eventData.Offset));
             Argument.AssertInRange(eventData.SequenceNumber, long.MinValue + 1, long.MaxValue, nameof(eventData.SequenceNumber));
-            Argument.AssertNotNull(context, nameof(context));
 
-            Logger.UpdateCheckpointStart(context.PartitionId, Identifier, EventHubName, ConsumerGroup);
+            Logger.UpdateCheckpointStart(partitionId, Identifier, EventHubName, ConsumerGroup);
 
             using var scope = EventDataInstrumentation.ScopeFactory.CreateScope(DiagnosticProperty.EventProcessorCheckpointActivityName);
             scope.Start();
 
             try
             {
-                // Parameter validation is done by Checkpoint constructor.
-
                 var checkpoint = new EventProcessorCheckpoint
                 {
                     FullyQualifiedNamespace = FullyQualifiedNamespace,
                     EventHubName = EventHubName,
                     ConsumerGroup = ConsumerGroup,
-                    PartitionId = context.PartitionId,
-                    StartingPosition = EventPosition.FromOffset(eventData.Offset)
+                    PartitionId = partitionId,
+                    StartingPosition = EventPosition.FromOffset(eventData.Offset, false)
                 };
 
-                return StorageManager.UpdateCheckpointAsync(checkpoint, eventData, cancellationToken);
+                await StorageManager.UpdateCheckpointAsync(checkpoint, eventData, cancellationToken).ConfigureAwait(false);
```

Comment:
just curious - why the change to await the task?

Reply:
The API was internal and we didn't feel the interim member was interesting in the stack trace and wouldn't surface directly to callers, but now that it is protected, I wanted to snap to our and prefer awaiting since I don't have strong justification to not do so.